Java plugin tests fail if you have a JDK newer than 17 but != 21

### Bug Description

If you have a JDK other than 21 that's newer than java 17, the tests fail.

### To Reproduce

```
$ dpkg -l|grep openjdk
ii  openjdk-17-jdk-headless:amd64                            17.0.14~6ea-1                            amd64        OpenJDK Development Kit (JDK) (headless)
ii  openjdk-17-jre-headless:amd64                            17.0.14~6ea-1                            amd64        OpenJDK Java runtime, using Hotspot JIT (headless)
ii  openjdk-21-jdk-headless:amd64                            21.0.6~6ea-1ubuntu2                      amd64        OpenJDK Development Kit (JDK) (headless)
ii  openjdk-21-jre-headless:amd64                            21.0.6~6ea-1ubuntu2                      amd64        OpenJDK Java runtime, using Hotspot JIT (headless)
ii  openjdk-25-jdk:amd64                                     25~4ea-1                                 amd64        OpenJDK Development Kit (JDK)
ii  openjdk-25-jdk-headless:amd64                            25~4ea-1                                 amd64        OpenJDK Development Kit (JDK) (headless)
ii  openjdk-25-jre:amd64                                     25~4ea-1                                 amd64        OpenJDK Java runtime, using Hotspot JIT
ii  openjdk-25-jre-headless:amd64                            25~4ea-1                                 amd64        OpenJDK Java runtime, using Hotspot JIT (headless)
```

```
pytest tests/unit/plugins/test_java_plugin.py::test_java_plugin_jre_21
```

### part yaml

```shell
n/a
```

### Relevant log output

```shell
$ pytest tests/unit/plugins/test_java_plugin.py::test_java_plugin_jre_21
========================================================================================= test session starts =========================================================================================
platform linux -- Python 3.11.10, pytest-8.3.4, pluggy-1.5.0
rootdir: /home/lengau/Work/Code/craft-parts
configfile: pyproject.toml
plugins: subprocess-1.5.3, cov-5.0.0, requests-mock-1.12.1, check-2.4.1, hypothesis-6.124.2, mock-3.14.0
collected 1 item

tests/unit/plugins/test_java_plugin.py F                                                                                                                                                        [100%]

============================================================================================== FAILURES ===============================================================================================
_______________________________________________________________________________________ test_java_plugin_jre_21 _______________________________________________________________________________________

part_info = <craft_parts.infos.PartInfo object at 0x7deb502fab90>, mocker = <pytest_mock.plugin.MockerFixture object at 0x7deb502e6510>

def test_java_plugin_jre_21(part_info, mocker):

orig_check_java = JavaPlugin._check_java

def _check_java(self, javac: str):
if "21" in javac:
return None, ""
return orig_check_java(self, javac)

mocker.patch.object(JavaPlugin, "_check_java", _check_java)

properties = MavenPlugin.properties_class.unmarshal({"source": "."})
plugin = MavenPlugin(properties=properties, part_info=part_info)
env = plugin.get_build_environment()
assert len(env) == 1
>       assert "17" in env["JAVA_HOME"]
E       AssertionError: assert '17' in '/usr/lib/jvm/java-25-openjdk-amd64'

/home/lengau/Work/Code/craft-parts/tests/unit/plugins/test_java_plugin.py:61: AssertionError
======================================================================================= short test summary info =======================================================================================
FAILED tests/unit/plugins/test_java_plugin.py::test_java_plugin_jre_21 - AssertionError: assert '17' in '/usr/lib/jvm/java-25-openjdk-amd64'
```

--- a/tests/unit/plugins/test_java_plugin.py
+++ b/tests/unit/plugins/test_java_plugin.py
@@ -43,12 +43,12 @@ def test_java_plugin_no_java(part_info, mocker):
     assert plugin.get_build_environment() == {}
 
 
-def test_java_plugin_jre_21(part_info, mocker):
+def test_java_plugin_jre_not_17(part_info, mocker):
 
     orig_check_java = JavaPlugin._check_java
 
     def _check_java(self, javac: str):
-        if "21" in javac:
+        if "17" not in javac:
             return None, ""
         return orig_check_java(self, javac)
 

